### ISSUE TITLE
Uncaught TypeError:Connot read property 'test' of undefined Chrome old versions <60

### ISSUE THREAD

**Issue by typeoo** [javascript-obfuscator/javascript-obfuscator#615]
## Expected Behavior
Hi,
I noticed that my obfuscated code doesn't work on the old versions of Chrome.
I tested it on most of old versions. (versions: 32,42,43,44 .... 63);
Also I tried to check the other browsers (old versions); I got the same problem from Opera too.
Firefox had no problem. 

![image](https://user-images.githubusercontent.com/66176156/83334553-a3919580-a2bc-11ea-896c-92979dd4467f.png)

it seems the problem is in this part:
        var _0x581498 = function() {
            var _0x50df5e = _0x581498[_0xa8ea('0x53e', '7]tY')](_0x2b610f[_0xa8ea('0x1b07', '%Rif')])()[_0xa8ea('0x12c8', 'KvLJ')](_0x2b610f[_0xa8ea('0x167a', 'Ci5&')]);
            return !_0x50df5e['test'](_0x5758cd);
        };
I'm trying to figure it out ,and also waiting your help. 
thanks for making this amazing tool.

**Comment by sanex3339** [comment]
Hi. Can you show me original source code of this part?

```
var _0x581498 = function() {
var _0x50df5e = _0x581498[_0xa8ea('0x53e', '7]tY')](_0x2b610f[_0xa8ea('0x1b07', '%Rif')])()_0xa8ea('0x12c8', 'KvLJ')]);
return !_0x50df5e'test';
};
```

**Comment by sanex3339** [comment]
Hmm.
Show me your obfuscator version, your options and example how I can reproduce this bug.

This is invalid code. How did you obfuscate your code?

**Comment by JerryChan31** [comment]
I bumped into this error, too.
In my case, I found this situation happens when you turned on `selfDefending` and `stringArray` at the same time.
This is my options and I use `webpack-obfuscator` to obfuscate my code.
```
const obfConfig = {
  compact: true,
  controlFlowFlattening: false,
  deadCodeInjection: false,
  debugProtection: false,
  debugProtectionInterval: false,
  disableConsoleOutput: true,
  identifierNamesGenerator: 'hexadecimal',
  log: false,
  renameGlobals: false,
  rotateStringArray: true,
  selfDefending: true,
  shuffleStringArray: true,
  splitStrings: true,
  stringArray: true,
  stringArrayEncoding: false,
  stringArrayThreshold: 1,
  unicodeEscapeSequence: false
}
```

**Comment by sanex3339** [comment]
Probably something other breaks your code, because here
```
var _0x581498 = function() {
    var _0x50df5e = _0x581498[_0xa8ea('0x53e', '7]tY')](_0x2b610f[_0xa8ea('0x1b07', '%Rif')])()_0xa8ea('0x12c8', 'KvLJ')]);
    return !_0x50df5e'test';
};
```

this line:
```
!_0x50df5e'test';
```

Is invalid JS code.

**Comment by sanex3339** [comment]
@JerryChan31 can you try to reproduce it here?
https://obfuscator.io/

**Comment by JerryChan31** [comment]
```js
var a0_0x1b976b = a0_0x9e3137(this, function() {
    var _0x13d134 = function() {
        var _0x38382d = _0x13d134[a0_0x3049('0x475')](a0_0x3049('0x4ac'))()[a0_0x3049('0x3f7')](a0_0x3049('0x438'));
        return !_0x38382d['test'](a0_0x1b976b); // errors here
    };
    return _0x13d134();
});
```
This is where my code throws error.
I can try to reproduce it later.

**Comment by typeoo** [comment]
hi Timofey
by turning off compact code I got it resolved. 
I think the problem is here: regular expression

**uncompacted code:**

`    if (!_0x51839a[_0x4add('0x9f4')]($, _0x4add('0x10a2'))[_0x4add('0x122d')]) {
        if (_0x51839a[_0x4add('0x25a')](_0x51839a[_0x4add('0xf3a')], _0x51839a[_0x4add('0xf3a')])) {
            return ![];
        } else {
            var _0x14e114 = /^[A-Za-z]*$/;
            if (_0x14e114['test'](result)) {
                return !![];
            } else {
                return ![];


            }
        }
    }`
I think regular expressions can throw an error on old versions when we compact the code:
`var english = /^[A-Za-z]*$/;
if (english.test(String.fromCharCode(result))){english_chr=true;}else{english_chr=false;}
if (result == result.toLowerCase())
{
 capslock=false;
}
else
{
 capslock=true;	
 result=result.toLowerCase();	
 
}`
* note I just had this problem on old versions.

**Comment by TerranTian** [comment]
Bumped into this problem too.

![image](https://user-images.githubusercontent.com/2083024/85641566-8389a200-b6c1-11ea-8abd-adc2d9941be1.png)

Here is the original code ：
`function loadLib(src, callback) {
        var s = document.createElement('script');
        s.async = false;
        s.src = src;
        s.addEventListener('load', function () {
            s.parentNode.removeChild(s);
            s.removeEventListener('load', arguments.callee, false);
            callback && callback();
        }, false);
        document.head.appendChild(s);
    };`

Here are the options:
`{
    "compact": true,
    "controlFlowFlattening": false,
    "deadCodeInjection": false,
    "debugProtection": false,
    "debugProtectionInterval": false,
    "disableConsoleOutput": false,
    "identifierNamesGenerator": "mangled",
    "log": false,
    "renameGlobals": false,
    "rotateStringArray": true,
    "selfDefending": true,
    "stringArray": true,
    "stringArrayEncoding": true,
    "stringArrayThreshold": 0.75,
    "unicodeEscapeSequence": false,

    "transformObjectKeys":false
}`

**Comment by sanex3339** [comment]
Just have applied for a free opensource BrowserStack plan, so can test it.

**Comment by sanex3339** [comment]
Found the problem.

**Comment by sanex3339** [comment]
Fixed in `1.2.1`
